cmd/dataavailability/data_availability_check.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/offchainlabs/nitro/arbstate"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/cmd/util/confighelpers"
+	"github.com/offchainlabs/nitro/das"
+	"github.com/offchainlabs/nitro/solgen/go/bridgegen"
+
+	flag "github.com/spf13/pflag"
+)
+
+// Data availability check is done to as to make sure that the data that is being stored by DAS is available at all time.
+// This done by taking the latest stored hash and an old stored hash (12 days) and it is checked if these two hashes are
+// present across all the DAS provided in the list, if a DAS does not have these hashes an error is thrown.
+// This approach does not guarantee 100% data availability, but it's an efficient and easy heuristic for our use case.
+//
+// This can be used in following manner (not an exhaustive list)
+// 1. Continuously call the function by exposing a REST API and create alert if error is returned.
+// 2. Call the function in an adhoc manner to check if the provided DAS is live and functioning properly.
+
+var sequencerInboxABI *abi.ABI
+var batchDeliveredID common.Hash
+var sequencerBatchDataABI abi.Event
+var addSequencerL2BatchFromOriginCallABI abi.Method
+
+const sequencerBatchDataEvent = "SequencerBatchData"
+const sequencerBatchDeliveredEvent = "SequencerBatchDelivered"
+
+type batchDataLocation uint8
+
+const (
+	batchDataTxInput batchDataLocation = iota
+	batchDataSeparateEvent
+)
```

Comment:
If you need these from the `das` package please make them public rather than copying the enum, which could easily get out of sync.

Reply:
Done.